java/src/main/java/io/cucumber/java/ObjectFactoryLoader.java

```diff
@@ -2,45 +2,55 @@
 
 import cucumber.api.java.ObjectFactory;
 import io.cucumber.core.io.ClassFinder;
-import io.cucumber.core.exception.CucumberException;
-import io.cucumber.core.reflection.NoInstancesException;
-import io.cucumber.core.reflection.Reflections;
-import io.cucumber.core.reflection.TooManyInstancesException;
 
-import static java.util.Arrays.asList;
+import java.util.Iterator;
+import java.util.ServiceLoader;
 
 public class ObjectFactoryLoader {
+
+    private static final ServiceLoader<ObjectFactory> LOADER = ServiceLoader.load(ObjectFactory.class);
+
     private ObjectFactoryLoader() {
     }
 
     /**
      * Loads an instance of {@link ObjectFactory}. The class name can be explicit, or it can be null.
-     * When it's null, the implementation is searched for in the <pre>cucumber.runtime</pre> packahe.
+     * When it's null, the implementation is searched for in the <pre>cucumber.runtime</pre> package.
      *
+     * @deprecated as of version 4.0.0; use {@code loadObjectFactory() } instead.
```

Comment:
While currently not package private, the ObjectFactoryLoader will be after merging. There is no need to keep a deprecated method around.